crates/infra/cli/src/commands/setup/mod.rs

```diff
@@ -37,9 +43,32 @@ impl OrderedCommand for SetupCommand {
         Terminal::step(format!("setup {name}", name = self.clap_name()));
 
         match self {
+            SetupCommand::ShellCompletions => generate_shell_completions(),
             SetupCommand::Cargo => setup_cargo(),
             SetupCommand::Npm => setup_npm(),
             SetupCommand::Pipenv => setup_pipenv(),
         }
     }
 }
+
+fn generate_shell_completions() -> Result<()> {
+    println!("Generating shell completions...\n");
+
+    let target_dir = std::env::var("CARGO_TARGET_DIR").unwrap_or_else(|_| "target".to_string());
```

Comment:
I suggest not using `CARGO_TARGET_DIR`, as that might change for different reasons. Since we always want to use the root `target` dir, even if `infra_cli` is built to somewhere different, I suggest using `Path::repo_path("target")` instead.

Reply:
Used the `Path::repo_path("target/shell-completions")`. I'm not a fan of hardcoding the `target` and for what it's worth the shell completions should probably be placed as outputs of the `infra_cli` build but leaving it for now and we can re-visit later, if needed.